src/System.Text.Json/source_package/README.md

```diff
@@ -0,0 +1,111 @@
+  - It is intentionally not easy to consume such a package since we wanted to avoid any tooling or packaging magic to predict the user's intention.
+    For simplicity, we recommended that you treat this like any other source file within your application or library.
+
+## Known Issues and Workarounds
+
+### Language Version
+
+* The `System.Text.Json` sources are built using C# 7.3 language features and hence you will likely see the following errors:
+  - `error CS8107: Feature 'default literal' is not available in C# 7.0. Please use language version 7.1 or greater.`
+  - `error CS8107: Feature 'ref structs' is not available in C# 7.0. Please use language version 7.2 or greater.`
+  - `error CS8320: Feature 'extensible fixed statement' is not available in C# 7.2. Please use language version 7.3 or greater.`
+
+* Therefore, it is recommended that you update your compiler to one that supports C# 7.3+ and add the following attribute to your project:
+  - `<LangVersion>latest</LangVersion>` OR `<LangVersion>7.3</LangVersion>`
+
+### Unsafe Code
+
+* The `System.Text.Json` sources are built using some unsafe code and hence you will likely see the following error:
+  - `error CS0227: Unsafe code may only appear if compiling with /unsafe`
+
+* Therefore, you will need the following attribute in your project (or something equivalent):
+  - `<AllowUnsafeBlocks>true</AllowUnsafeBlocks>`
+
+### Missing Package References
+
+* The `System.Text.Json` sources depend on the following packages:
+  - `System.Memory`
+  - `System.Runtime.CompilerServices.Unsafe`
+  - `System.Buffers`
+  - `System.Numerics.Vectors`
+
+* You will likely see errors like the following when you don't have the required packages referenced in your application:
+  - `error CS0103: The name 'Unsafe' does not exist in the current context`
+  - `error CS0246: The type or namespace name 'ReadOnlySpan<>' could not be found (are you missing a using directive or an assembly reference?)`
+
+* Therefore, you will need to add the following package references in your project:
+  - `<PackageReference Include="System.Runtime.CompilerServices.Unsafe" Version="..." />`
+  - `<PackageReference Include="System.Memory" Version=..." />`
+  - The System.Memory package transitively brings in the other dependencies.
+
+### CLS Compliance
+
+* The `System.Text.Json` sources are built as part of a CLS Compliant library. Hence, they contain the `CLSCompliant` attributes in certain places.
+  You might see warnings like the following (or errors if warnings are treated as errors):
+  - `warning CS3021: 'Utf8JsonWriter.WriteNumber(string, ulong, bool)' does not need a CLSCompliant attribute because the assembly does not have a CLSCompliant attribute`
+
+* You could either mark your project as CLSCompliant, or opt-out of this particular warning, or manually modify the sources to remove that attribute where it is used:
```

Comment:
I wouldn't advice people to change the files from the source package. The reason is that the sources all live in the user's package cache, which means two things: 1) Changes won't persist when building the same project in a different machine or when nuget cache moves or is deleted. 2) Changes will apply to every single app that uses the source package on the same machine, which might get in the way for other apps that do need to be cls compliant.